lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -4773,16 +4776,20 @@ getLazyBitcodeModuleImpl(std::unique_ptr<MemoryBuffer> &&Buffer,
                          LLVMContext &Context, bool MaterializeAll,
                          DiagnosticHandlerFunction DiagnosticHandler,
                          bool ShouldLazyLoadMetadata = false) {
-  BitcodeReader *R =
-      new BitcodeReader(Buffer.get(), Context, DiagnosticHandler);
```

Comment:
The problem was here. The BitcodeReader will create a std::unique_ptr around the MemoryBuffer that gets passed in, but we don't release the original unique_ptr until the end of the function, so there are two owners during the execution of `getBitcodeModuleImpl`, which throws an exception.